src/main/java/hu/bme/mit/spaceship/GT4500.java

```diff
@@ -74,23 +87,20 @@ public boolean fireTorpedo(FiringMode firingMode) {
             // if both of the stores are empty, nothing can be done, return failure
           }
         }
-        break;
-
-      case ALL:
-        // try to fire both of the torpedo stores
-        //TODO implement feature
-	if(!primaryTorpedoStore.isEmpty()){
-		firingSuccess = primaryTorpedoStore.fire(1);
-		wasPrimaryFiredLast = true;
-	}
-	if(!secondaryTorpedoStore.isEmpty()){
-		firingSuccess = secondaryTorpedoStore.fire(1);
-		wasPrimaryFiredLast = false;	
-	}
-        break;
-    }
-
-    return firingSuccess;
+	  return firingSuccess;
+  }
+  
+  private boolean allFire(){
```

Comment:
Nagyon tetszik, ahogy ennyire részletes az elemzés, még arra is kitér, hogy üres-e lövés előtt.